components/src/menu/menu-trigger-for.directive.ts

```diff
@@ -0,0 +1,108 @@
+import { Overlay, OverlayConfig, OverlayRef } from "@angular/cdk/overlay";
+import { TemplatePortal } from "@angular/cdk/portal";
+import { Directive, ElementRef, Input, OnDestroy, ViewContainerRef } from "@angular/core";
+import { Observable, Subscription } from "rxjs";
+import { filter, mergeWith } from "rxjs/operators";
+
+import { MenuComponent } from "./menu.component";
+
+@Directive({
+  selector: "[bitMenuTriggerFor]",
+  host: {
+    role: "button",
+    "aria-haspopup": "menu",
+    "[attr.aria-expanded]": "isOpen",
+    "(click)": "toggleMenu()",
+  },
+})
+export class MenuTriggerForDirective implements OnDestroy {
+  private isOpen = false;
+  private overlayRef: OverlayRef;
+  private defaultMenuConfig: OverlayConfig = {
+    panelClass: "bit-menu-panel",
+    hasBackdrop: true,
+    backdropClass: "cdk-overlay-transparent-backdrop",
+    scrollStrategy: this.overlay.scrollStrategies.reposition(),
+    positionStrategy: this.overlay
+      .position()
+      .flexibleConnectedTo(this.elementRef)
+      .withPositions([
+        {
+          originX: "start",
+          originY: "bottom",
+          overlayX: "start",
+          overlayY: "top",
+        },
+      ])
```

Comment:
Do we want this to be configurable? This positioning will probably work well in most cases, but if we have an item on the right side of the screen that opens a menu, we would want it to pin to the right side and open left. The account switcher in desktop is an example of what I'm thinking of.

Reply:
We can probably do this by supplying 
`{
      originX: "end",
      originY: "bottom",
      overlayX: "end",
      overlayY: "top",
    }` as the second option in the ConnectedPositions array. If the first one doesn't fit in the view well, the second one should then be chosen by angular cdk. 
Or we can make it an input and let the dev decide where they want it (bottom left, bottom right, etc).

Reply:
Providing it as an alternative position in the array is a good solution for now. We can add an `@Input` for different/additional positions if required later.

Reply:
Update: this is done